### PR TITLE
fix: swap completed event not firing

### LINKED ISSUE #9138: MetaMetrics missing Swap Completed event
### What is this about?

Data team noticed the `Swap Completed` is not received for mobile.
We receive the `Swap Started` event.

### Scenario

```gherkin
GIVEN the app is installed and configured with a wallet
AND users has tokens to swap
WHEN user swaps tokens using MM swap feature
AND swap completes suceesfully
THEN a "Swap Completed" event should appear for this user on Segment source
AND on Mixpanel after a few seconds
```

### Design

_No response_

### Technical Details

We migrated from Mixpanel SDK to Segment SDK recently.
However, the only part of the add that throws the event is https://github.com/MetaMask/metamask-mobile/blob/main/app/components/Nav/Main/RootRPCMethodsUI.js#L213-L220 and was not modified by this migration. However the `trackSwaps` function in RootRPCMethodsUI.js was modified to use the new `useMetrics()` hook `trackAnonymousEvent` function.
We have to investigate this.

### Threat Modeling Framework

_No response_

### Acceptance Criteria

- have the `Swap Completed` event visible in Segment source (and then Mixpanel) on swap completion.

### Stakeholder review needed before the work gets merged

- [X] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by infiniteflower**
## **Description**

This PR fixes the `SWAP_COMPLETED` event not firing after a transaction has been confirmed.

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-mobile/issues/9138

## **Manual testing steps**

1. Do a Smart Transaction swap 
2. Wait for tx to be confirmed
3. Observe `TRACK event saved {"event": "Swap Completed"` in the terminal

Repeat the above with a non Smart Transaction swap

## **Screenshots/Recordings**

n/a 

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by NicolasMassart — DISMISSED** [review]
Looks good to me.

File: app/components/Nav/Main/index.test.tsx [hunk 1]
```diff
@@ -14,4 +19,114 @@ describe('Main', () => {
     const wrapper = shallow(<MainAppContainer />);
     expect(wrapper).toMatchSnapshot();
   });
+
+  describe('useSwapConfirmedEvent', () => {
+    it('queues transactionMeta ids correctly', () => {
+      const txMetaId = 'foo';
```
Comment (NicolasMassart) [inline app/components/Nav/Main/index.test.tsx:25]: is this the actual format of ids? random chars string?

File: app/components/Nav/Main/index.test.tsx [hunk 2]
```diff
@@ -14,4 +19,114 @@ describe('Main', () => {
...
+      const eventEmitter = new EventEmitter();
+
+      const { result } = renderHook(() =>
+        useSwapConfirmedEvent({
+          TransactionController: {
+            hub: eventEmitter,
+          },
+          swapsTransactions: {
+            [txMetaId]: {
+              analytics: 'hello',
```
Comment (NicolasMassart) [inline app/components/Nav/Main/index.test.tsx:54]: S: using values that are actualy possible in the contect of the project helps using the test as a doc to understand what the code does. Here hello world doesn't provide this ability.

File: app/components/Nav/Main/index.test.tsx (hunk 1 above)
Comment (infiniteflower) [reply app/components/Nav/Main/index.test.tsx]: Updated to make test data more realistic, it's a uuid

File: app/components/Nav/Main/index.test.tsx (hunk 2 above)
Comment (infiniteflower) [reply app/components/Nav/Main/index.test.tsx:90]: Updated to make test data more realistic

**Review by NicolasMassart — APPROVED** [review]
Looks good to me.

**Pull request merged**